### DISCUSSION TITLE
Is there a way to use PipeListenerOptions instance_limit of 1?

### DISCUSSION THREAD

**Discussion by DeweertD** [kotauskas/interprocess#75] [Q&A]
Hello!  Thank you so much for such an incredible library! 

[ I am using 2.2.1 with tokio ]

I am working on a windows project and was wondering if there was a way to implement an instance limit of 1?  I don't see any configuration for using ListenerOptions + Streams, so looking at the pure-windows PipeListenerOptions + DuplexPipeListener led me to notice [the comment about the limit](https://github.com/kotauskas/interprocess/blob/main/src/os/windows/named_pipe/listener/options.rs#L40-L45) needing to be somewhere between 2 and 254 (specifically it says a value of 1 breaks .accept(), but I don't see a way of making a connection without using .accept())

Would you know of a method besides accept which would allow the limit to be enforced?  Or maybe some functional workaround that lets me run my own accept / limit the server side to a single client connection?

Thank you for any of your time!

**Comment by kotauskas** [comment]
I'm planning to add support for listener-less use of Windows named pipes (i.e. closer to the real Windows API) sometime in the future. I'm currently not sure how `instance_limit` = 1 could be made to work with the listener-oriented local socket API, but for the Windows-specific side of things, there is definitely a path forward (albeit one that entails a considerable amount of refactoring).
